pytorch_lightning/callbacks/base.py

```diff
@@ -148,10 +148,22 @@ def on_predict_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningMo
         """Called when the predict epoch ends."""
 
     def on_epoch_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        """Called when either of train/val/test epoch begins."""
+        r"""
+        .. deprecated:: v1.6
+            This callback hook was deprecated in v1.6 and will be removed in v1.8. Use
+            ``on_<train/validation/test>_epoch_start`` instead.
```

Comment:
thinking that this won't be shown to the user if we remove a reference from docs

Reply:
the references will still be there if they search for it or go to API references.

Reply:
https://147528-178626720-gh.circle-artifacts.com/0/html/api/pytorch_lightning.callbacks.base.html#pytorch_lightning.callbacks.base.Callback.on_epoch_start